### PR TITLE
[B5] Updated styling for invalid select2

### PR DESCRIPTION
**Pull request by orangejenny**
## Product Description
https://dimagi.atlassian.net/browse/QA-7209

This came up during QA for locations B5, but I didn't realize until after merging [that PR](https://github.com/dimagi/commcare-hq/pull/35286).

For invalid select2s, the new B5 icon overlaps with the X to clear the selection, which also makes that X not clickable.
Screenshot 2024-11-05 at 9 01 29 AM

After this PR, here are the states of this widget:
### Valid, no selection
Screenshot 2024-11-05 at 4 26 56 PM

### Valid, option selected
Screenshot 2024-11-05 at 4 26 50 PM

### Invalid, no selection
Screenshot 2024-11-05 at 8 54 55 AM

### Invalid, option selected
Screenshot 2024-11-05 at 8 55 05 AM

I debated a little what order to put the X and icon, and decided to put the icon in the middle because it separates the two clickable things (X and arrow) more definitively. It's also a simpler approach, since the icon and arrow are already in the same element.

### Safety story
Styling change to some very specific rules. No effect beyond this widget.

### Automated test coverage

no

### QA Plan

QA will verify the fix, although I'm not blocking merge on the verification.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by nospame — APPROVED** [review]
Change looks good to me. Are the "valid" screenshots here from B3? The layout and dropdown arrow are different from the "invalid" screenshots.

File: corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss [hunk 1]
```diff
@@ -110,9 +110,9 @@
   }
   .select2-selection--single .select2-selection__arrow {
     background-repeat: no-repeat;
-    background-size: 24%;
-    background-position: 15px 7px;
-    width: 66px !important;
+    background-size: 32%;
```
Comment (biyeun) [inline corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss:113]: can you cross check your changes with the select2s here? I spent a lot of time adjusting these values to fit well https://staging.commcarehq.org/styleguide/b5/molecules/selections/

**Review by biyeun — CHANGES_REQUESTED** [review]
@orangejenny can you do before/after of the valid screenshots? I see the down arrow looks different

File: corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss (hunk 1 above)
Comment (biyeun) [reply corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss:113]: ok i see what this is doing. why the decision to make the icon smaller? I adjusted it specifically to match the other bootstrap5 fields. can we do away with these changes?

**Comment by orangejenny** [comment]
Fixed the screenshots - the first two were indeed from B3, I got my wires crossed in my local environment.

File: corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss:113]: I can make it bigger, it's not intentionally smaller. I was just eyeballing it.

Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_select2.scss:113]: Here are better screenshots, I don't think it's smaller. (I'm opening these images in separate tabs and flipping between them.)

|Screenshot 2024-11-05 at 4 36 08 PM|Screenshot 2024-11-05 at 4 36 05 PM|
|---|---|

**Comment by biyeun** [comment]
@orangejenny I decided to poke around with this and fix the "hack" of using the arrow container as the background container for the validation state icon. instead, I'm using the `::before` pseudo class. I also fixed the state for `is-valid` in addition to `is-invalid` and made sure the icon size for the validation icon remained the same (to match the regular select validation icon). I also added a clearable select2 to the styleguide so that styles could be cross checked there

here is a non-clearable single-option select2:
Screenshot 2024-11-05 at 11 10 38 PM
Screenshot 2024-11-05 at 11 11 10 PM
Screenshot 2024-11-05 at 11 10 56 PM

here is a clearable single-option select2:
Screenshot 2024-11-05 at 11 12 41 PM
Screenshot 2024-11-05 at 11 13 05 PM
Screenshot 2024-11-05 at 11 12 56 PM

what it looks like when you click "x" (and the `is-invalid` class remains):
Screenshot 2024-11-05 at 11 13 35 PM

**Review by biyeun — APPROVED** [review]

**Comment by orangejenny** [comment]
@biyeun Nice, thank you - and thank you for adding style guide examples.

@nospame Could you review this? I can't approve because I'm the author, and Biyeun's approval apparently doesn't count because she's the last pusher.

**Review by nospame — APPROVED** [review]

**Pull request merged**